packages/nvidia_nat_mcp/src/nat/plugins/mcp/cli/commands.py

```
@@ -454,24 +464,70 @@ async def ping_mcp_server(url: str,
-    from mcp.client.streamable_http import streamablehttp_client
 
     async def _ping_operation():
-        # Select transport
+        if auth_redirect_uri:
+            # Auth-enabled path: use WorkflowBuilder so OAuth2 is negotiated
+            try:
+                from nat.builder.workflow_builder import WorkflowBuilder
+                from nat.plugins.mcp.client.client_config import MCPServerConfig
+            except ImportError:
+                raise RuntimeError("MCP client functionality requires nvidia-nat-mcp package. "
+                                   "Install with: uv pip install nvidia-nat-mcp")
```

Ruff (0.15.11)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)